### ISSUE TITLE
Installation troubles

### ISSUE THREAD

**Issue by kevit** [swiftstack/swift_undelete#1]
Hi guys. Here is not about really issues, but mostly lack of documentation and my middleware loading process misunderstanding.

What I expect: 
I did installation of middleware using setup.py

Also I add a configuration settings to /etc/swift/proxy-server.conf

[pipeline:main]
pipeline = swift_undelete

[filter:swift_undelete]
paste.filter_factory = swift_undelete:filter_factory

Swift proxy booting properly, but when I tried delete a object nothing happened. I expect new .trash- container. 

What exactly I did wrong?

**Comment by limga** [comment]
0. download source from github and install it "python setup.py install"
1. follow configuration write your proxy-server.conf  and insert swift_undelete in pipeline
2. swift-init proxy object-expirer start.
3. Delete any object. it move to  trash container. will be removed by object-expirer.

-------------------------------------
[filter:swift_undelete]
use = egg:swift_undelete#undelete
trash_prefix = .trash-
trash_lifetime = 120
block_trash_deletes = no
enable_by_default = yes

**Comment by jiasir** [comment]
Hi guys! Here is my configuration of `proxy-server.conf`
```
[pipeline:main]
pipeline = swift_undelete

[filter:swift_undelete]
use = egg:swift_undelete#undelete
trash_prefix = .trash-
trash_lifetime = 120
block_trash_deletes = no
enable_by_default = yes
```

When I restart proxy, I got the following error:
```
LookupError: No section 'swift_undelete' (prefixed by 'app' or 'application' or 'composite' or 'composit' or 'pipeline' or 'filter-app') found in config /etc/swift/proxy-server.conf
```

When I change the `[filter:swift_undelete]` to `[filter-app:swift_undelete]`, I got another error:
```
LookupError: The [filter-app:swift_undelete] section in /etc/swift/proxy-server.conf is missing a 'next' setting
```
And how to fix it? Thanks!

**Comment by tipabu** [comment]
Swift uses [PasteDeploy](https://bitbucket.org/ianb/pastedeploy/) to construct WSGI applications. While [PasteDeploy's homepage](http://pythonpaste.org/deploy/) seems to be down currently, [other](http://docs.pylonsproject.org/projects/pyramid/en/latest/narr/paste.html) [places](docs.repoze.org/moonshining/tools/paste.html) include some docs, and the whole site is still available [via the Wayback Machine](https://web.archive.org/web/20161013165027/http://pythonpaste.org/deploy/) if you want to learn more about pipelines, filters, and apps.

The TL;DR though is that the filter needs to be *added* to the proxy pipeline, rather than *replacing* it. I pushed up 764c8b8 which should help explain that.

**Comment by jiasir** [comment]
@tipabu Many thanks! it worked!
